ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/rollup/RollupCreatePayload.java

```diff
@@ -0,0 +1,197 @@
+    if (parentBlock.isEmpty()) {
+      return replyWithStatus(requestId, RollupCreatePayloadStatus.INVALID_TERMINAL_BLOCK);
+    }
+
+    try {
+      final PayloadCreationResult result =
+          mergeCoordinator.createBlock(
+              parentBlock.get(), timestamp, suggestedRecipient, transactions, prevRandao);
+
+      if (result.getBlockValidationResult().errorMessage.isPresent()) {
+        return new JsonRpcSuccessResponse(
+            requestId,
+            new RollupCreatePayloadResult(
+                RollupCreatePayloadStatus.INVALID_BLOCK,
+                result.getBlockValidationResult().errorMessage));
+      }
+
+      final BlockCreationResult blockCreationResult = result.getBlockCreationResult();
+      final EngineGetPayloadResult payloadResult =
+          blockResultFactory.enginePayloadTransactionComplete(blockCreationResult.getBlock());
+
+      final List<RollupCreatePayloadResult.InvalidTransactionResult> invalidTransactionResults =
+          invalidTransactionResults(
+              rawTransactions, transactions, blockCreationResult.getTransactionSelectionResults());
+      final List<String> unprocessedTransactions =
+          unprocessedTransactions(
+              rawTransactions,
+              transactions,
+              blockCreationResult.getBlock().getBody().getTransactions(),
+              blockCreationResult.getTransactionSelectionResults().getInvalidTransactions().stream()
+                  .map(TransactionValidationResult::getTransaction)
+                  .collect(Collectors.toList()));
+
+      return new JsonRpcSuccessResponse(
+          requestId,
+          new RollupCreatePayloadResult(
+              RollupCreatePayloadStatus.PROCESSED,
+              result.getPayloadId(),
+              payloadResult,
+              invalidTransactionResults,
+              unprocessedTransactions));
+    } catch (Exception e) {
+      LOG.error("Failed to create block: ", e);
+      return new JsonRpcErrorResponse(requestId, JsonRpcError.INTERNAL_ERROR);
+    }
+  }
+
+  private ArrayList<String> unprocessedTransactions(
+      final List<?> requestedTransactionsRaw,
+      final List<Transaction> requestedTransactions,
+      final List<Transaction> includedTransactions,
+      final List<Transaction> invalidTransactions) {
+    final ArrayList<String> unprocessedTransactions =
+        new ArrayList<>(
+            requestedTransactions.size()
+                - includedTransactions.size()
+                - invalidTransactions.size());
+
+    for (int i = 0; i < requestedTransactionsRaw.size(); i++) {
+      Transaction tx = requestedTransactions.get(i);
+      if (!includedTransactions.contains(tx) && !invalidTransactions.contains(tx)) {
+        unprocessedTransactions.add((String) requestedTransactionsRaw.get(i));
+      }
+    }
+
+    return unprocessedTransactions;
+  }
+
+  private List<RollupCreatePayloadResult.InvalidTransactionResult> invalidTransactionResults(
```

Comment:
Both of these methods should probably live in a rollup transaction results class. Grouping the transactions isn't really an RPC specific function.

Reply:
After 2nd thought, decided to move that logic `RollupCreatePayloadResult` as it's tight to that result itself.